lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+		average = probeData.runningTotal / count;
+	} else average = 0; 
+	
+	var probeEvent = {
+		"count" : count,
+		"minimum" : probeData.min,
+		"maximum" : probeData.max,
+		"average" : average
+	}
+	
+	// Reset values
+	probeMetrics[probeName] = undefined;
+	
+	return probeEvent;
+}
+
+function getHeapData() {
+	var jsonToReturn = heapMetrics;
+
+	// Reset JSON
+	heapMetrics = {};
+
+	return jsonToReturn
+}
+
+function getGCHeapData() {
+	if (gcMetrics.count > 0) {
+		var usedHeap = gcMetrics.used / gcMetrics.count;
+	
+		//Reset values
+		gcMetrics.used = 0;
+		gcMetrics.count = 0;
+	} else {
+		usedHeap = -1;
+	}
+	return usedHeap;
+}
+
+function getMessageData() {
+    if (messageMetrics.sent != undefined && messageMetrics.received != undefined) {
+        var oldMessageMetrics = messageMetrics;
+        messageMetrics = {};
+        return oldMessageMetrics;
+    }
+    return null;	
+}
+
+function compareHistograms(oldHistogram, newHistogram) {
+	var objectCounts = {};
+
+	// Add all the instances currently in the heap.
+	for ( var type in newHistogram) {
+		var newTuple = newHistogram[type];
+		var oldTuple = oldHistogram[type];
+		var diffTuple;
+		if (!oldTuple) {
+			diffTuple = newTuple;
+		} else {
+			diffTuple = {
+				'count' : newTuple['count'] - oldTuple['count'],
+				'size' : newTuple['size'] - oldTuple['size'],
+			};
+		}
+		objectCounts[type] = diffTuple;
+	}
+
+	// Account for any types where all instances have been deleted.
+	for ( var type in oldHistogram) {
+		var oldTuple = oldHistogram[type];
+		var diffTuple = objectCounts[type];
+		if (!diffTuple) {
+			diffTuple = {
+				'count' : 0 - oldTuple['count'],
+				'size' : 0 - oldTuple['size'],
+			};
+			objectCounts[type] = diffTuple;
+		}
+	}
+
+	return objectCounts;
+}
+
+function initialise(){
+	// Initialise
+	createEventLoopListener();
+    createCpuListener();
+    createHeapListener();
+    createGCListener();
+    createMessageListener();
+    createHTTPListener();
+	createObjectListener();
+	
+	for(i in probeList){
+		// Create listeners for all probes
+		createProbeListener(probeList[i]);
+	}
+}
+// ---------------------------------------------
+
+function updateMetrics(probeName, value){
+	
+	var metrics = probeMetrics[probeName];
+	
+	// update metrics accordingly (and if they haven't been defined yet)
+	if (!metrics.min || value < metrics.min){
+		metrics.min = value;
+	}
+	if (!metrics.max || value > metrics.max){
+		metrics.max = value;
+	}
+	
+	// Update other values accordingly
+	metrics.runningTotal += value;
+	metrics.count++;
+}
+
+// Export the desired functions
+var metrics = {
+	'startCpuProfiling' : startCpuProfiling,
+	'stopCpuProfiling' : stopCpuProfiling,
+	'startTrackingObjects' : startTrackingObjects,
+	'stopTrackingObjects' : stopTrackingObjects
+};
+
+var config = {
+	hostname : os.hostname(),
+	appName : path.basename(process.argv[1]),
+	key : 'testkey'
+};
+
+var internal = {
+		'on' : internal_on
+};
+
+
+module.exports.use = use;
```

Comment:
`exports.use = use;` (`module.` is redundant), and place at top of file, underneath the requires, so its easy to see at a glance what a module exports.